### DISCUSSION TITLE
Question: does enabling threads on uniprocessor machine (i.e., a VPS) actually helps with performance?

### DISCUSSION THREAD

**Discussion by catfluoride** [searx/searx#3067] [General]
Hi all,

sorry if this is not the right place where to ask, and feel free to redirect me elsewhere.

I host SearX on a cheap uniprocessor (single cpu, single core) VPS on Hetzner. In this situation, there are real benefits in enabling threading for uwsgi or it just adds overhead?

Thanks a lot in advance for your help.

**Comment by unixfox** [comment]
@dalf maybe you will be able to answer that

**Comment by dalf** [comment]
On a small VPS, one thing to take care is the memory: each uwsgi worker takes about 256MB (with searx/searx, if you want to proxy the images, use Morty). By default the number of worker is 4, so it should be okay with CX11 of Hetzner (2Go of RAM) as long the server don't host other services.

I don't really know the answer to your question: with one core it makes sense to set the number of worker to 1 and increase the number of the threads, but the Python GIL may decrease the response time.

You can try:
* to set the number of workers to 1
* to set the number of threads to 4 

And compare the response time and load average to the default configuration.

---

You can get the response time per engine in FF/Chrome Dev Tools (F12):
![image](https://user-images.githubusercontent.com/1594191/142406271-3a9d99be-fcaa-4fd4-8c06-567299942ebd.png)

Check with the multiple queries.

**Reply by catfluoride** [reply]
Thanks a lot for the detailed response!

Indeed, my VPS is CX11 and SearX is the only service on it. I'll experiment a bit with dev tools and different configs, as you have suggested.
